### PR TITLE
Add HTTP request and response data viewer

### PR DESCRIPTION
**Pull request by jdongyun**
### Describe your change:
This code allows you to view requests and responses made during HTTP web communication directly without going through a web browser.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x]  This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: web_programming/http_request_and_response.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    request(sock, req_type="POST", endpoint="/", headers=headers, body=body)
+
+
+def response(sock: socket.socket) -> None:
+    """
+    Print received data
+
+    :param sock: socket client
+    :return:
+    """
```
Comment (CaedenPH) [inline web_programming/http_request_and_response.py:53]:
```suggestion
    """
```

File: web_programming/http_request_and_response.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+
+def request_json_data(sock: socket.socket, body: str) -> None:
+    """
+    Request json style data
+
+    :param sock: socket client
+    :param body: json encoded string
+    :return:
+    """
```
Comment (CaedenPH) [inline web_programming/http_request_and_response.py:40]:
```suggestion
    """
```

File: web_programming/http_request_and_response.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    Request HTTP connection
+
+    :param sock: socket client
+    :param req_type: request type (ex: GET, POST)
+    :param endpoint: endpoint
+    :param headers: http headers
+    :param body: http body
+    :return:
+    """
```
Comment (CaedenPH) [inline web_programming/http_request_and_response.py:17]:
```suggestion
    """
```

File: web_programming/http_request_and_response.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    sock: socket.socket, req_type: str, endpoint: str, headers: list[str], body: str
+) -> None:
+    """
+    Request HTTP connection
+
+    :param sock: socket client
+    :param req_type: request type (ex: GET, POST)
+    :param endpoint: endpoint
+    :param headers: http headers
+    :param body: http body
```
Comment (CaedenPH) [inline web_programming/http_request_and_response.py:15]:
```suggestion
```
I'm not sure if these are very useful, the variable names should be self documenting enough

**Review by CaedenPH — CHANGES_REQUESTED** [review]
Perhaps you could link some documentation about HTTP, and the different methods/protocols

**Comment by cclauss** [comment]
See CONTRIBUTING.md.  This is a tool, not an algorithm.  Algorithmic functions do not print.

**Pull request closed without merging**